core/src/main/java/io/grpc/internal/Http2ClientStream.java

```diff
@@ -49,21 +50,30 @@
   /**
    * Metadata marshaller for HTTP status lines.
    */
-  private static final Metadata.AsciiMarshaller<Integer> HTTP_STATUS_LINE_MARSHALLER =
-      new Metadata.AsciiMarshaller<Integer>() {
+  private static final InternalMetadata.TrustedAsciiMarshaller<Integer> HTTP_STATUS_MARSHALLER =
+      new InternalMetadata.TrustedAsciiMarshaller<Integer>() {
         @Override
-        public String toAsciiString(Integer value) {
-          return value.toString();
+        public byte[] toAsciiString(Integer value) {
+          throw new UnsupportedOperationException();
         }
 
+        /**
+         * RFC 7231 says status codes are 3 digits long.
+         *
+         * @see: <a href="https://tools.ietf.org/html/rfc7231#section-6">RFC 7231</a>
+         */
         @Override
-        public Integer parseAsciiString(String serialized) {
-          return Integer.parseInt(serialized.split(" ", 2)[0]);
+        public Integer parseAsciiString(byte[] serialized) {
+          if (serialized.length >= 3) {
+            return (serialized[0] - '0') * 100 + (serialized[1] - '0') * 10 + (serialized[2] - '0');
```

Comment:
I agree, but I think it is much faster to read this way.  There is no doubt that it is correct.